### ISSUE TITLE
400 rest_invalid_param -> rest_additional_properties_forbidden

### ISSUE THREAD

**Issue by landwire** [WordPress/gutenberg#31018]
## Description
I am a plugin developer and I am extending all core blocks with an ability to hide them on certain device width: https://wordpress.org/plugins/responsive-block-control

A user recently reported the following error, when previewing certain blocks, which I could reproduce on calendar, archive, latest comments:

`{"code":"rest_invalid_param","message":"Invalid parameter(s): attributes","data":{"status":400,"params":{"attributes":"responsiveBlockControl is not a valid property of Object."},"details":{"attributes":{"code":"rest_additional_properties_forbidden","message":"responsiveBlockControl is not a valid property of Object.","data":null}}}}`

Is this a bug within Gutenberg? Why does it only happen on some blocks?

The code I am using:

```
const {
			responsiveBlockControl,
		} = attributes;

		const handleChange = ( device ) => {
			const newValue = ! responsiveBlockControl[ device ];

			delete responsiveBlockControl[ device ];

			const blockOptions = Object.assign( { [ device ]: newValue }, responsiveBlockControl );

			dispatch( 'core/block-editor' ).updateBlockAttributes( clientId, { responsiveBlockControl: blockOptions } );
		};
```

## Step-by-step reproduction instructions
1. Add any additional property to all blocks (or just use my plugin to do that)
2. Go to insert a block and choose one of those: calendar or archive
3. You will see an error in the console and the preview pane with a message block cannot previewed

## Expected behaviour
Proper preview

## Actual behaviour
Broken JS

## WordPress information
- WordPress version: 5.7.1
- Gutenberg version: whichever ist installed with 5.7.1
- Are all plugins except Gutenberg deactivated? Gutenberg is not installed. It is vanilla wordpress
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes, Twenty Twenty One

## Device information
- Device: 
- Operating system: 
- Browser:

**Comment by landwire** [comment]
Is there any feedback on this possible soonish? It has been 2 months since I asked this question and obviously would like to resolve this for my plugin users.

**Comment by tellthemachines** [comment]
This is happening for all blocks that use [Server Side Render](https://github.com/WordPress/gutenberg/blob/trunk/packages/server-side-render/README.md#L1), because the API request from that component includes the block attributes as query parameters. These get compared to a schema [that excludes any non-default block attributes](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/endpoints/class-wp-rest-block-renderer-controller.php#L68), causing that error.

We could fix this by filtering the attributes that get added to the query path in ServerSideRender to only include the default block attributes.

**Comment by Mamaduka** [comment]
The issue was fixed in WP 5.8. See #29197.

Like, @tellthemachines suggested attributes are now sanitized before sending the request.
